base/boot.jl

```diff
@@ -242,6 +242,9 @@ ccall(:jl_toplevel_eval_in, Any, (Any, Any),
       (f::typeof(Typeof))(x) = ($(_expr(:meta,:nospecialize,:x)); isa(x,Type) ? Type{x} : typeof(x))
       end)
 
+macro Typeof(x)
+    _expr(:Typeof, _expr(:escape, x))
```

Comment:
This should probably not use `escape`, but be handled specially in macro expansion, like `symbolicgoto`: https://github.com/JuliaLang/julia/blob/7d117dceb0f791e9ab9a7d5864e32809c3ce16c0/src/macroexpand.scm#L356